src/topic-client.ts

```diff
@@ -0,0 +1,56 @@
+import {PubsubClient} from './internal/pubsub-client';
+import {TopicPublish, MomentoLogger} from '.';
+import {range} from './internal/utils/collections';
+import {TopicClientProps} from './topic-client-props';
+
+/**
+ * Momento Topic Client.
+ *
+ * Publish and subscribe to topics.
+ */
+export class TopicClient {
+  private readonly logger: MomentoLogger;
+  private readonly pubsubClients: Array<PubsubClient>;
+  private nextPubsubClientIndex: number;
+
+  /**
+   * Creates an instance of TopicClient.
+   */
+  constructor(props: TopicClientProps) {
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    this.logger.info('Creating Momento CacheClient');
+
+    const numClients = 1;
```

Comment:
I ported this from the cache client. This may be a premature feature.

Reply:
yeah this won't help with the subscribe calls since they will have a stream on a single client/channel; let's just do this as a single client for now.

Reply:
Removed in [991dd1c](https://github.com/momentohq/client-sdk-nodejs/pull/347/commits/991dd1cd7e447357896c8a8c06810e26318b0c27)